src/TLuaInterpreter.cpp

```diff
@@ -7094,6 +7137,157 @@ int TLuaInterpreter::permSubstringTrigger( lua_State * L )
     return 1;
 }
 
+int TLuaInterpreter::permKey( lua_State *L )
+{
+    string luaNameID = "";
+    if( ! lua_isstring( L, 1 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #1 type (key name as string expected, got %1!)" )
+            .arg( luaL_typename( L, 1 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaNameID = lua_tostring( L, 1 );
+    }
+
+    string luaParent = "";
+    if( ! lua_isstring( L, 2 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #2 type (key parent group as string expected, got %1!)" )
+            .arg( luaL_typename( L, 2 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaParent = lua_tostring( L, 2 );
+    }
+
+    int luaKeyCode;
+    if( ! lua_isstring( L, 3 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #3 type (key code as string expected, got %1!)" )
+            .arg( luaL_typename( L, 3 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaKeyCode = lua_tointeger( L, 3 );
+    }
+
+    int luaModifier;
+    if( ! lua_isstring( L, 4 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #4 type (key modifier as string expected, got %1!)" )
+            .arg( luaL_typename( L, 4 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaModifier = lua_tointeger( L, 4 );
+    }
+
+
+    string luaFunction = "";
+    if( ! lua_isstring( L, 5 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #5 type (key lua code as string expected, got %1!)" )
+            .arg( luaL_typename( L, 5 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaFunction = lua_tostring( L, 5 );
+    }
+
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "permKey: NULL Host pointer - something is wrong!" )
+            .toUtf8().constData() );
+        return 2;
+    }
+    TLuaInterpreter * pLuaInterpreter = pHost->getLuaInterpreter();
+    QString _luaNameID = luaNameID.c_str();
+    QString _luaParent = luaParent.c_str();
+    QString _luaFunction = luaFunction.c_str();
+    int _luaKeyCode = luaKeyCode;
+    int _luaModifier = luaModifier;
+    int keyID = pLuaInterpreter->startPermKey( _luaNameID, _luaParent, _luaKeyCode, _luaModifier, _luaFunction );
+    lua_pushnumber( L, keyID );
+    return 1;
+}
+
+int TLuaInterpreter::tempKey( lua_State *L )
+{
+    int luaKeyCode;
+    if( ! lua_isstring( L, 1 ) )
+    {
+        lua_pushstring( L, tr( "tempKey: bad argument #1 type (key code as string expected, got %1!)" )
```

Comment:
So I've got a game plan for attacking the lua key name export, just have to figure out a good way to automate the separation of mod keys into their own lua list, instead of lumping them all into a single list, as they are represented by the existing QMap.

Would it make sense to just put all the keycodes into a single list, and let the functions that use those values figure it out at runtime, or is it important that there is both a mudlet.keys list and a mudlet.keymods list?

Whichever you think is better, I'll jump onto tomorrow (Saturday), and then I think it's just making the keymods an optional parameter, resolving the couple conflicts that've popped up, and I think that should be it, barring any further changes you'd like to see.